packages/pwa-buildpack/lib/WebpackTools/PWADevServer.js

```diff
@@ -19,7 +19,17 @@ const addImgOptMiddleware = require('../Utilities/addImgOptMiddleware');
 
 // We only need the HTTP server to detect if the host is SSL or not.
 const HttpsServer = require('https').Server;
-const isHttpsServer = server => server instanceof HttpsServer;
+const isHttpsServer = server => {
+    return (
+        process.env.DEV_SERVER_SSL_OFFLOADED || server instanceof HttpsServer
+    );
+};
+
+// If SSL offloaded, then decorate existing listening port for user interaction
+const getDecoratedListeningPort = port => {
+    const publicPort = process.env.DEV_SERVER_SSL_OFFLOADED_PORT || 443;
+    return process.env.DEV_SERVER_SSL_OFFLOADED ? publicPort : port;
+};
 
```

Comment:
I think that what we're looking for here is already supported by [`devServer.publicPath`](https://v4.webpack.js.org/configuration/dev-server/#devserverpublicpath-). You should be able to set it to a full URL, as the documentation says, which could include a port.

I think we made a mistake and forgot to make `output.publicPath` and `devServer.publicPath` configurable with environment variables. (I think it was there before, actually)

My suggestion is this: let's define another environment variable called `DEV_SERVER_PUBLIC_PATH` , which can be `"/"` by default. Then, we'll use that value as the [`devServer.publicPath`](https://v4.webpack.js.org/configuration/dev-server/#devserverpublicpath-), and use it as the [`output.publicPath`](https://v4.webpack.js.org/configuration/output/#outputpublicpath) as well if we're in watch mode. This way we can simply use our env var configuration system to expose the functionality that Webpack already has, and documents.

Then, in this callback, where we are trying to log the dev server URL, the line which tries `url = new URL(webpackDevServerOptions.publicPath).href` would work!